config/src/main/java/no/sysco/middleware/kafka/clients/interceptor/config/ConfigHarvesterInterceptor.java

```diff
@@ -0,0 +1,173 @@
+package no.sysco.middleware.kafka.clients.interceptor.config;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerInterceptor;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerInterceptor;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.streams.StreamsConfig;
+
+/**
+ * Kafka interceptor for harvesting and storing Producer/Consumer/KafkaStreams user provided configs
+ *
+ * @author SYSCO Middleware
+ * @param <K>
+ * @param <V>
+ */
+public class ConfigHarvesterInterceptor<K, V> implements ProducerInterceptor<K, V>, ConsumerInterceptor<K, V> {
+
+    /**
+     * configurable property for this interceptor that may be used to override default target topic
+     */
+    public static final String TOPIC_NAME = "config.harvester.topic";
+
+    private static final String DEFAULT_TOPIC_NAME = "__client_configs";
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    private String topicName = DEFAULT_TOPIC_NAME;
+
+    //we need to remove interceptors or we are going into endless interception loop
+    private static final String[] IGNORED_PROPERTIES = {
+        ProducerConfig.INTERCEPTOR_CLASSES_CONFIG,
+        ConsumerConfig.INTERCEPTOR_CLASSES_CONFIG,
+        StreamsConfig.PRODUCER_PREFIX + ProducerConfig.INTERCEPTOR_CLASSES_CONFIG,
+        StreamsConfig.CONSUMER_PREFIX + ConsumerConfig.INTERCEPTOR_CLASSES_CONFIG};
+
+    private static final String[] FILTERED_CONFIG_VALUES = new String[]{
+        "ssl.key.password", "ssl.keystore.password"
+    };

```

Comment:
Eventually, this could be also configurable, as ignored props, but let's just keep it as it is.

Reply:
ok